[seriouslia0] iP

seriouslia0 opened the pull request:
### ChadGPT

> “Your mind is for holding ideas, not having them.” – Satthew Mimon ([Not from this source](https://dansilvestre.com/productivity-quotes))

ChadGPT is a extreme productivity tool to take you out of the matrix

- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use
- All you need to do is,
- 

1. download it from([here](https://nus-cs2103-ay2223s2.github.io/website/schedule/week4/project.html). double-click it.)
2. add your tasks.
3. let it manage your tasks for you 😉
4. And it is FREE!


Features:

- [x] 1.  Managing tasks
- [x] 2.  Managing deadlines 
- [ ] 3.  Reminders 
- [ ] 4. Helping you become the AI Chad you always wanted to be

If you Java programmer, you can use it to practice Java too. Here's Floyd warshall :



```
class FloydWarshall {
  final static int INF = 9999, nV = 4;

  // Implementing floyd warshall algorithm
  void floydWarshall(int graph[][]) {
    int matrix[][] = new int[nV][nV];
    int i, j, k;

    for (i = 0; i < nV; i++)
      for (j = 0; j < nV; j++)
        matrix[i][j] = graph[i][j];

    // Adding vertices individually
    for (k = 0; k < nV; k++) {
      for (i = 0; i < nV; i++) {
        for (j = 0; j < nV; j++) {
          if (matrix[i][k] + matrix[k][j] < matrix[i][j])
            matrix[i][j] = matrix[i][k] + matrix[k][j];
        }
      }
    }
    printMatrix(matrix);
  }

  void printMatrix(int matrix[][]) {
    for (int i = 0; i < nV; ++i) {
      for (int j = 0; j < nV; ++j) {
        if (matrix[i][j] == INF)
          System.out.print("INF ");
        else
          System.out.print(matrix[i][j] + "  ");
      }
      System.out.println();
    }
  }

  public static void main(String[] args) {
    int graph[][] = { { 0, 3, INF, 5 }, { 2, 0, INF, 4 }, { INF, 1, 0, INF }, { INF, INF, 2, 0 } };
    FloydWarshall a = new FloydWarshall();
    a.floydWarshall(graph);
  }
}
```

Commit:
Add Gradle support

seriousbo committed:
Commit for Level-2

seriousbo committed:
level 3

seriousbo committed:
Revert "level 3"

This reverts commit c92b9c2f839299389362b75f2db7e1c30dfb1dd9.

seriousbo committed:
level 3

seriousbo committed:
Revert "Revert "level 3""

This reverts commit 162cbd1f9e4ca50f867c8d5083ab80ea489710fd.

seriousbo committed:
changes to to level-4

seriousbo committed:
level 5

seriousbo committed:
Level 6

seriousbo committed:
Added TaskList class to make my oop more upz

seriousbo committed:
Added InvalidDescriptionException

seriousbo committed:
Added Save feature

kennethk-1201 commented on src/main/java/CommandManager.java:
```diff
@@ -0,0 +1,111 @@
+        return;
+    }
+
+    public static void addTodo(String input, ArrayList<Task> storedTask, int taskCounter) {
+        String[] newInput = input.split("todo ");
+        storedTask.add(taskCounter, new Todo(newInput[1]));
+        taskCounter++;
+        System.out.println("Gotchu fam");
+        System.out.printf("I've added\n" + storedTask.get(taskCounter-1).toString() +
+                "\nto all the shit u need to do\n");
+        System.out.printf("shag bro now u got %d tasks\n", taskCounter);
+        return;
+    }
```
This function is beautiful.

kennethk-1201 reviewed (approved):
LGTM

seriousbo committed:
Level-8

seriousbo committed:
Fixed everything

seriousbo committed:
Merge commit '48bc58be134170fb207f69a8aebc190353f18649'

# Conflicts:
#	src/main/java/CommandManager.java
#	src/main/java/Deadline.java
#	src/main/java/Event.java

daitenshionyan commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,37 @@
+public class Task {
+
+    String name;
+    boolean completionStatus;
```
Perhaps a more "boolean" name here? Maybe `isCompleted`?

daitenshionyan commented on src/main/java/Todo.java:
```diff
@@ -0,0 +1,16 @@
+import java.util.*;
```
It seems this import was not used, is this import really needed?

daitenshionyan commented on src/main/java/TaskList.java:
```diff
@@ -0,0 +1,121 @@
+        System.out.println("Gotchu fam");
+        System.out.printf("I've added\n" + tasks.get(currentSize-1).toString() +
+                "\nto all the shit u need to do\n");
+        System.out.printf("shag bro now u got %d tasks\n", currentSize);
```
I like how you broke the lines out, for other lines as well. It made visualising a lot easier!!

daitenshionyan commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,39 @@
+            }
+        }
+
+        System.out.println("Thanks for using me brother");
```
I like how the logical components are split up here!!

daitenshionyan commented on src/main/java/DateTimeParser.java:
```diff
@@ -0,0 +1,61 @@
+import java.time.format.DateTimeParseException;
+
+//remember to fix this shit and account for all the different ways that people can fuck up entering the date
+//only can handle date in DD/MM/YYYY HHHH
```
I love this reminder comment.

daitenshionyan reviewed (commented):
I like your coding style line spacings. Just some nits which CheckStyle should be able to spot. Bot has a solid personality too.

seriousbo committed:
Add more oop minimal goal

seriousbo committed:
Refactor everything into packages it was a nightmare and i broke everything this module is exposing me so hard

Statspadders commented on src/main/java/duke/helper/Parser.java:
```diff
@@ -0,0 +1,37 @@
+//this is the class that parses commands
+public class Parser {
+
+    public static void run(String input, String[] command, TaskList tasks) {
+        try {
+            if (command[0].equals("list")) {
+                tasks.listTasks();
+            } else if (command[0].equals("mark")) {
+                tasks.mark(command);
+            } else if (command[0].equals("unmark")) {
+                tasks.unmark(command);
+            } else if (command[0].equals("todo")) {
+                tasks.addTask(input);
+            } else if (command[0].equals("deadline")) {
+                tasks.addDeadline(input);
+            } else if (command[0].equals("event")) {
+                tasks.addEvent(input);
+            } else if (command[0].equals("delete")) {
+                tasks.deleteTask(command);
+            } else {
+                System.out.println("Invalid command wake up brother");
+            }
+        } catch (InvalidTaskDescriptionException e) {
+            System.out.println(e.getMessage());
+        }
```
A suggestion : Consider using the switch expression, it might increase readability and code quality !

Statspadders commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,51 @@
+package duke;// Tan Matthew Simon Castaneda
+
+import duke.helper.Parser;
+import duke.storage.Storage;
+import duke.tasks.TaskList;
+import duke.ui.Ui;
```
Great import statements. Clear to the reader on what Java packages you are using, helps improve the code quality 👍

Statspadders commented on src/main/java/duke/tasks/TaskList.java:
```diff
@@ -0,0 +1,139 @@
+    public ArrayList<Task> getTasks() {
+        return this.tasks;
+    }
+
+
+
+
```
Consider removing redundant spaces in between codes to improve the code quality 😄

Statspadders commented on src/main/java/duke/tasks/TaskList.java:
```diff
@@ -0,0 +1,139 @@
+
+
+    //delete
+    public void deleteTask(String[] command) {
+        int taskPointer = Integer.parseInt(command[1]) - 1;
+        //duke.tasks.Task temp = duke.tasks.get(taskPointer);
+        try {
+            Task temp = tasks.get(taskPointer);
+            tasks.remove(taskPointer);
+            System.out.println("Ok bro I've removed this : ");
+            System.out.printf(temp.toString() +
+                    "\nfrom all the shit u need to do\n");
+            System.out.println("Hope you have a better life now");
+        } catch (IndexOutOfBoundsException e) {
+            System.out.println("brother look at how long ur list is first then delete lah");
+        }
+    }
```
Consider moving the system.out.println() section to the UI.java. To increase the code quality and to let the users focus on major operations of the TaskList class.

Statspadders reviewed (commented):
LGTM, almost. Just a few nits to fix.
Overall, I enjoyed reading your code, it was entertaining.

seriousbo committed:
Merge commit '556af3f47a96b32898ab4cdbd65b16486a4871e8'

seriousbo committed:
Edited build.gradle file to correct filepath

seriousbo committed:
Added unit test for Todo class

seriousbo committed:
Added unit test for Todo class

seriousbo committed:
Merge commit '3ebe54a80911ab61a16bc7d9dbc7201055d923db'

seriousbo committed:
Add JavaDocs to make my comments more upz

seriousbo committed:
Made code compliant to coding standards

seriousbo committed:
Add find functionality

seriousbo committed:
Added more JavaDocs because i am upz

seriousbo committed:
Merge branch 'A-CodingStandard' into A-JavaDoc

seriousbo committed:
Merge branch 'Level-9' into A-JavaDoc

seriousbo committed:
Add JavaFX

seriousbo committed:
no message

seriousbo committed:
Add assertion statements

seriousbo committed:
Edited code according to coding quality standards

seriouslia0 committed:
Merge pull request #1 from seriouslia0/branch-A-CodeQuality

Edited code according to coding quality standards

seriouslia0 committed:
Merge pull request #2 from seriouslia0/branch-A-Assertions

Add assertion statements

seriousbo committed:
Add contacts feature to ChadGPT

seriousbo committed:
Added the user guide.

seriousbo committed:
Updated existing javadocs

seriousbo committed:
Named level 9 branch wrongly so i renamed it

seriousbo committed:
Named level 10 branch wrongly so I renamed it

seriousbo committed:
Currently many violations in coding style and missing javadocs.

Makes some of the code difficult to read.

Following proper coding style and documentation will make the code more readable and scalable.

Let's,
* fix all the coding style errors
* add all the missing javadocs

seriousbo committed:
no message

seriousbo committed:
Fixed methods to be shorter
Fixed too long nesting
Handled exceptions in Parser Class

seriouslia0 committed:
Update MainWindow.java
